src/main/java/net/snowflake/ingest/streaming/internal/DataValidationUtil.java

```diff
@@ -37,12 +45,26 @@ class DataValidationUtil {
   public static final int BYTES_8_MB = 8 * 1024 * 1024;
   public static final int BYTES_16_MB = 2 * BYTES_8_MB;
 
+  static final int MAX_VARIANT_LENGTH = BYTES_16_MB - 1024; // SNOW-664249
```

Comment:
Please add TODO keyword, move the comment to the prev line and add some description about the JIRA
```suggestion
  // TODO SNOW-664249: [add your description]
  static final int MAX_VARIANT_LENGTH = BYTES_16_MB - 1024;
```

Reply:
And I feel like 1024 is too much, have you try with 32 or 64?

Reply:
Seems to be working fine with 64.